### ISSUE TITLE
Nextcloud - 503 error

### ISSUE THREAD

**Issue by v0lv0x** [buttercup/buttercup-desktop#324]
Hi all!
Just to bring an issue.
I had a functional archive on a Nextcloud server using the desktop app. But after a fresh install of Xubuntu, I have systematically 503 errors when trying to save a new pass and randomly 503 errors when trying to open the archive.

**Comment by morojgovany** [comment]
If you are on linux I have a "temporary fix". You can connect to a webdav provider typing in the file manager davs://your.cloud.host/ (I think you should mount it in your FM before trying to connect it). 
In the buttercup desktop app you can now add a "local" archive pointing to your webdav.
Sadly it is not possible in the browser extension.
Hope it helps, I took a look on the code but it's too hard for my dev low level !

**Comment by v0lv0x** [comment]
It worked! Thank you!

**Comment by perry-mitchell** [comment]
Hi @v0lv0x @morojgovany - Thanks for looking at this. We need to setup a dev instance of Nextcloud for future work, which should happen this weekend (hopefully). Providing that goes well, we'll look into this straight away.

**Comment by morojgovany** [comment]
You can open a nextcloud account on this provider [https://cloud.indie.host/apps/registration/](https://cloud.indie.host/apps/registration/) it's free for very limited capacity (50Mo) and hosted in France with respect for privacy policy (CHATONS). [https://chatons.org/en/chatons-le-collectif-anti-gafam](https://chatons.org/en/chatons-le-collectif-anti-gafam) (in french)

Details : 
- [https://indie.host/about/](https://indie.host/about/)
- [https://indie.host/](https://indie.host/)  

Thank's !

**Comment by perry-mitchell** [comment]
Great idea, thanks! I think we want to also have a host within a sub folder to maximise our testing with regards to different environments. Otherwise this setup may be similar to the demo accounts on the Nextcloud site. Will try this nonetheless..

**Comment by perry-mitchell** [comment]
Got a nextcloud instance running and was able to reproduce straight away:

![image](https://user-images.githubusercontent.com/3869469/29994601-e1856bf6-8fdb-11e7-9bc5-1239f84d0d40.png)

Error occurred immediately when saving a new entry. After then closing the archive and trying to log in again, a similar error occurs:

![image](https://user-images.githubusercontent.com/3869469/29994606-16211d38-8fdc-11e7-9cf0-b9d1bd5d16ad.png)

@sallar I'll raise this to critical I think.

**Comment by sallar** [comment]
@perry-mitchell I tried requesting the exact same request that Buttercup is doing but in Paw app, and it works. So the remote server is rejecting Buttercup for some reason, maybe referral headers or something? I will try to investigate

**Comment by sallar** [comment]
@perry-mitchell Found it. This cookie: `ocgrd6wpkg2l=d3fe89236f8c2af....;` is causing the request to return `503` and fail. I don't know where it comes from. Any ideas?

**Comment by sallar** [comment]
@perry-mitchell Alright. Came closer to the issue. `node-fetch` is used in `webdav-client` (and `webdav-fs` and in result in Webdav, Owncloud and Nextcloud adapters), and that fails in electron while requesting the nextcloud endpoint because of that cookie which Nextcloud sets in first response. But `window.fetch` doesnt fail for some reason.

**Comment by perry-mitchell** [comment]
> node-fetch is used in webdav-client (and webdav-fs and in result in Webdav, Owncloud and Nextcloud adapters)

Good point! Well shall we try changing the fetch adapter? Buttercup core supports that now:

```javascript
const Buttercup = require("buttercup");
Buttercup.vendor.webdavFS.setFetchMethod(window.fetch);
```

or

```javascript
import Buttercup from "buttercup-web";
Buttercup.vendor.webdavFS.setFetchMethod(window.fetch);
```

That might help with the cookie. We still get problems regarding this in the browser extension. It's because the browser keeps and sends cookies regardless of what we try. Need to look into that a bit more.

**Comment by mannp** [comment]
I have this error today on a recently installed buttercup 0.25.2-1 on arch linux.

I created an archive on nextcloud through the connect cloud sources option and have been using it for a couple of weeks without issue.

Today I try and connect to the archive through the same server and I am getting this error 503 each time.

I am able to open the file locally and the password works fine.

I am also able to create a new archive on the same nextcloud server and connect and use the archive as normal.

I did have a corrupt file on my letsencrypt proxy, so had to recover my nextcloud nginx config from backup, but that is the only change I can think of that could have any baring on this.

Thanks for any ideas to resolve.
